File: frontend/src/api/policy.ts
import apiClient from './client'

export interface PolicyDto {
  policyType: 'TERMS' | 'PRIVACY'
  version: string
  content: string
  effectiveDate: string
}

export interface CurrentPoliciesDto {
  terms: PolicyDto | null
  privacy: PolicyDto | null
}

export const policyApi = {
  getCurrentPolicies: async (): Promise<CurrentPoliciesDto> => {
    const response = await apiClient.get<CurrentPoliciesDto>('/policies/current')
    return response.data
  },

  getPolicy: async (type: 'TERMS' | 'PRIVACY'): Promise<PolicyDto> => {
    const response = await apiClient.get<PolicyDto>(`/policies/${type.toLowerCase()}`)
    return response.data
  }
}

File: ios/Dutypark/Core/Models/Policy.swift
import Foundation

struct PolicyDto: Decodable, Identifiable {
    let policyType: String
    let version: String
    let content: String
    let effectiveDate: String

    var id: String { "\(policyType)-\(version)" }
}

struct CurrentPoliciesDto: Decodable {
    let terms: PolicyDto?
    let privacy: PolicyDto?
}
